### PR TITLE
fix(ipfs-topology): Don't swarm connect to ourselves if we are in the peerlist

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

File: packages/ipfs-topology/src/ipfs-topology.ts [hunk 1]
```diff
@@ -111,8 +111,16 @@ export class IpfsTopology {
     ipfs: IpfsApi,
     bootstrapList: string[]
   ): Promise<void> {
+    // Don't want to swarm connect to ourself
+    const myPeerId = (await ipfs.id()).id
+    const filteredBootstrapList = bootstrapList.filter((addr) => {
+      const arr = addr.split('/')
+      const peerId = arr[arr.length - 1]
+      return myPeerId != peerId
```
Comment (v-stickykeys) [inline packages/ipfs-topology/src/ipfs-topology.ts:119]: Can replace this with `endsWith` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith

Comment (v-stickykeys) [inline packages/ipfs-topology/src/ipfs-topology.ts:119]:
```suggestion
      return !addr.endsWith(myPeerId)
```

**Review by v-stickykeys — APPROVED** [review]
👍 small suggestion

**Review by ukstv — CHANGES_REQUESTED** [review]
+1 to @v-stickykeys

File: packages/ipfs-topology/src/ipfs-topology.ts (hunk 1 above)
Comment (stbrody) [reply packages/ipfs-topology/src/ipfs-topology.ts:119]: oh yeah, that's much nicer, thanks!

**Pull request merged**